server/services/store/sqlstore/blocks.go

```diff
@@ -245,6 +245,37 @@ func (s *SQLStore) insertBlock(db sq.BaseRunner, block *model.Block, userID stri
 		return err
 	}
 
+	if existingBlock == nil && (block.Type == "image" || block.Type == "attachment") {
+		fileIDsToRestore := make([]string, 0, 2)
+
+		if fileIDVal, exists := block.Fields["fileId"]; exists {
+			if fileIDStr, ok := fileIDVal.(string); ok && fileIDStr != "" {
+				fileID := retrieveFileIDFromBlockFieldStorage(fileIDStr)
+				if fileID != "" {
+					fileIDsToRestore = append(fileIDsToRestore, fileID)
+				}
+			}
+		}
+
+		if attachmentIDVal, exists := block.Fields["attachmentId"]; exists {
```

Comment:
Since we migrated away from `attachmentId`, we should no longer have any code to continue supporting it. If we still think some users have `attachmentId`, we can add a migration to update it. Otherwise we're basically supporting two ways of doing the same thing and we'd have to always keep supporting both keys.